ray-on-gke/cloudbuild.yaml

```diff
@@ -26,34 +34,32 @@ steps:
         terraform apply -var=project_id=$PROJECT_ID \
         -var=cluster_name=ray-$SHORT_SHA-$_PR_NUMBER-cluster \
         -var=region=$_ZONE -auto-approve -no-color \
-        || ( terraform destroy -auto-approve && exit 1 )
+        || ( exit 1 )
+        echo "pass" > /workspace/gke_cluster_result.txt
     dir: 'gke-platform/'
-    waitFor: ['validate platform', 'validate user']
-
-  - id: 'get kube config'
-    name: 'gcr.io/cloud-builders/gcloud'
-    entrypoint: 'sh'
-    args: 
-    - '-c'
-    - |
-      gcloud container clusters get-credentials \
-      ray-$SHORT_SHA-$_PR_NUMBER-cluster \
-      --zone $_ZONE \
-      --project $PROJECT_ID 
-    waitFor: ['create gke cluster']
+    allowFailure: true
+    waitFor: ['validate platform', 'validate user', 'validate jupyterhub']
   
-  - id: 'create ray cluster'
+  - id: 'test ray cluster'
     name: 'gcr.io/$PROJECT_ID/terraform'
     entrypoint: 'sh'
     args:
       - '-c'
       - |
+        # Get kube config
+        gcloud container clusters get-credentials \
+        ray-$SHORT_SHA-$_PR_NUMBER-cluster \
+        --zone $_ZONE \
+        --project $PROJECT_ID 
+
+        cd ray-on-gke/user/
         terraform apply \
         -var=project_id=$PROJECT_ID \
         -var=namespace=$SHORT_SHA \
         -var=service_account=$_USER_NAME-$SHORT_SHA-system-account \
         -auto-approve -no-color \
-        || ( echo "false" > /workspace/user_result )
+        || ( exit 1 )
```

Comment:
is the `|| ( exit 1)` step still needed if we have `allowFailure: true`?

Reply:
Good catch